driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -0,0 +1,125 @@
+import org.bson.BsonDocument;
+import org.bson.BsonDouble;
+import org.bson.BsonInt32;
+import org.bson.BsonInt64;
+import org.bson.BsonNull;
+import org.bson.BsonString;
+import org.bson.BsonValue;
+import org.bson.conversions.Bson;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+/**
+ * Convenience methods related to {@link Expression}.
+ */
+public final class Expressions {
+
+    private Expressions() {}
+
+    /**
+     * Returns an expression having the same boolean value as the provided
+     * boolean primitive.
+     *
+     * @param of the boolean primitive
+     * @return the boolean expression
+     */
+    public static BooleanExpression of(final boolean of) {
+        // we intentionally disallow ofBoolean(null)
+        return new MqlExpression<>((codecRegistry) -> new BsonBoolean(of));
+    }
+
+    /**
+     * Returns an expression having the same integer value as the provided
+     * int primitive.
+     *
+     * @param of the int primitive
+     * @return the integer expression
+     */
+    public static IntegerExpression of(final int of) {
+        return new MqlExpression<>((codecRegistry) -> new BsonInt32(of));
+    }
+    public static IntegerExpression of(final long of) {
+        return new MqlExpression<>((codecRegistry) -> new BsonInt64(of));
+    }
+    public static NumberExpression of(final double of) {
+        return new MqlExpression<>((codecRegistry) -> new BsonDouble(of));
+    }
+    public static DateExpression of(final Instant of) {
+        return new MqlExpression<>((codecRegistry) -> new BsonDateTime(of.toEpochMilli()));
+    }
+
+    /**
+     * Returns an expression having the same string value as the provided
+     * string.
+     *
+     * @param of the string
+     * @return the string expression
+     */
+    public static StringExpression of(final String of) {
+        return new MqlExpression<>((codecRegistry) -> new BsonString(of));
+    }
+
+    /**
+     * Returns an array expression containing the same boolean values as the
+     * provided array of booleans.
+     *
+     * @param array the array of booleans
+     * @return the boolean array expression
+     */
+    public static ArrayExpression<BooleanExpression> ofBooleanArray(final boolean... array) {
+        List<BsonValue> result = new ArrayList<>();
+        for (boolean b : array) {
+            result.add(new BsonBoolean(b));
+        }
+        return new MqlExpression<>((cr) -> new BsonArray(result));
+    }
+
+
+    public static ArrayExpression<IntegerExpression> ofIntegerArray(final int... ofIntegerArray) {
+        List<BsonValue> array = Arrays.stream(ofIntegerArray)
+                .mapToObj(BsonInt32::new)
+                .collect(Collectors.toList());
+        return new MqlExpression<>((cr) -> new BsonArray(array));
+    }
+
+    public static DocumentExpression ofDocument(final Bson document) {
```

Comment:
The names of all expression methods and helper methods should be reviewed for consistency.

Reply:
1. The return types of methods should be reviewed. (consider `<T extends DocumentEx>`)
2. Review all comments in https://github.com/mongodb/mongo-java-driver/pull/1023

Reply:
For 1 above, confirm whether "schema objects" would extend DocEx.

Reply:
Make set and unset match in their return: https://github.com/mongodb/mongo-java-driver/pull/1052#discussion_r1031945379